test/types/client_bulk_write.test-d.ts

```diff
@@ -0,0 +1,258 @@
+import { expectError, expectType } from 'tsd';
+
+import {
+  type ClientBulkWriteModel,
+  type ClientDeleteManyModel,
+  type ClientDeleteOneModel,
+  type ClientInsertOneModel,
+  type ClientReplaceOneModel,
+  type ClientUpdateManyModel,
+  type ClientUpdateOneModel,
+  type Document,
+  type Filter,
+  type MongoClient,
+  type OptionalId,
+  type UpdateFilter,
+  type UUID,
+  type WithoutId
+} from '../mongodb';
+
+declare const client: MongoClient;
+type Book = { title: string; released: Date };
+type Author = { name: string; published: number };
+type Store = { _id: UUID };
+
+// Baseline check that schema modifies the following fields for each type.
+expectType<ClientInsertOneModel<Book>['document']>(null as unknown as OptionalId<Book>);
```

Comment:
`expectType` has the signature of `expectType<T>(expression: T)` - I think these are asserting backwards (as if we did `expect(expected).to.deep.equal(actual)`).  It took me a second to realize these are asserting that `ClientInsertOneModel<Book>['Document']` is of type `OptionalId<Book>`.

Usually we'd do something like:

```typescript
{ // check that the schema modifies the following fields for each type
  declare const model: ClientInsertOneModel<Book>;
  expectType<OptionalId<Book>(model.document);
  ...
}
```